supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystem.scala

```diff
@@ -198,8 +205,25 @@ trait SharedFileSystem extends PathFactory {
       PairOfFiles(src, dest)
     }
 
+    // A possibly staged version of the input file suitable for downstream processing, or just the original input
+    // file if no staging was required.
+    val staged: WomFile = value.mapFile { input =>
+      pathBuilders.collectFirst({ case http: HttpPathBuilder if HttpPathBuilder.accepts(input) => http }) match {
+        case Some(http) =>
+          implicit val materializer = ActorMaterializer()
+          implicit val ec = actorContext.dispatcher
+          val stagedFile = better.files.File.newTemporaryFile()
+          val futurePath = http.contentAsByteIterator(input) map { bytes =>
+            stagedFile.writeBytes(bytes)
+            stagedFile.pathAsString
+          }
+          Await.result(futurePath, Duration.Inf)
```

Comment:
Should I be giving the side eye here or is there a reason it's AOK? (including "good enough for now!"

Reply:
The other network path builders also block on localization, they just hide that better behind their NIO interfaces. 🙂

Reply:
🙈